cypress/support/commands.js

```diff
@@ -12,59 +30,162 @@ import { AUTO_SAVE_PAUSE, LOAD_PAGE_PAUSE } from '../constants/constants';
+);
+
+Cypress.Commands.add('saveAnswer', () => {
+  cy.get(SAVE_INPUT_BUTTON).click();
+});
+
+Cypress.Commands.add(
+  'editSettings',
+  ({ headerVisibility, showAutomaticFeedback, hint, numAttemptsAllowed }) => {
+    cy.get(SETTINGS_BUTTON).click();
+
+    cy.get(SETTINGS_MODAL).should('be.visible');
+
+    // change headerVisibility if set
+    if (_.isBoolean(headerVisibility)) {
+      cy.get(HEADER_VISIBILITY_SWITCH).then(($headerVisible) => {
+        // header visible
+
+        const isHeaderVisibilityChecked = $headerVisible.attr('checked');
+        // turn on
+        if (headerVisibility && isHeaderVisibilityChecked !== 'checked') {
+          cy.get(HEADER_VISIBILITY_SWITCH).click();
+        }
+        // turn off
+        else if (!headerVisibility && isHeaderVisibilityChecked === 'checked') {
+          cy.get(HEADER_VISIBILITY_SWITCH).click();
+        }
+      });
     }
 
-    const input = cy.get(inputTextField);
-    if (!text || text === '') {
-      input.clear().should('be.empty');
-    } else {
-      input.type(text).contains(text);
+    // change showAutomaticFeedback if set
+    if (_.isBoolean(showAutomaticFeedback)) {
+      cy.get(SHOW_AUTOMATIC_FEEDBACK_SWITCH).then(($showAutomaticFeedback) => {
+        // is automatic feedback set
+        const showAutomaticFeedbackChecked = $showAutomaticFeedback.attr(
+          'checked'
+        );
+        // turn on
+        if (
+          showAutomaticFeedback &&
+          showAutomaticFeedbackChecked !== 'checked'
+        ) {
+          cy.get(SHOW_AUTOMATIC_FEEDBACK_SWITCH).click();
+        }
+        // turn off
+        else if (
+          !showAutomaticFeedback &&
+          showAutomaticFeedbackChecked === 'checked'
+        ) {
+          cy.get(SHOW_AUTOMATIC_FEEDBACK_SWITCH).click();
+        }
+      });
     }
 
-    // wait is necessary to let the data be sent to the api
-    if (online) {
-      cy.wait(AUTO_SAVE_PAUSE);
+    // set hint
+    if (_.isString(hint)) {
+      cy.get(HINT_INPUT).clear();
+      if (hint.length) {
+        cy.get(HINT_INPUT).type(hint);
+      }
     }
-  }
-);
 
-Cypress.Commands.add(
-  'clearStudentResponse',
-  (loadPage = true, online = true) => {
-    cy.enterStudentResponse('', loadPage, online);
+    // set attempts
+    if (_.isNumber(numAttemptsAllowed)) {
+      cy.get(NUM_ATTEMPTS_ALLOWED_INPUT).then((input) => {
+        const value = toInteger(input.attr('value'));
+        // console.log('value: ', value);
+        // input.setAttribute('value', 20);
+        const newValue = value - numAttemptsAllowed;
+        if (newValue !== 0) {
+          const valueToType = Array.from(
+            { length: numAttemptsAllowed - 1 },
+            () => '{uparrow}'
+          ).join();
+
+          cy.get(NUM_ATTEMPTS_ALLOWED_INPUT)
+            .clear()
+            .type(valueToType)
+            .trigger('change');
+        }
+      });
+    }
+    // save
+    cy.get(SAVE_SETTINGS_BUTTON).click({ force: true });
```

Comment:
What does `force: true` do?

Reply:
It clicks without throwing an error even if the element is not clickable. (I'll add this comment)